#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 5
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "robot"
    }
  ]
}
Viewpoint {
  orientation -0.046208834441383614 -0.9965393780055789 0.06909422337520392 5.429446701765944
  position 0.3816084760705981 0.07794732941052003 0.5051081290162011
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Robot {
  translation -0.224154 0.02 0.0596071
  children [
    SharpGP2Y0A02YK0F {
      translation 0.055 -0.03 0.09
      rotation -0.5773509358560258 0.577349935856137 -0.577349935856137 -2.094395307179586
      name "irleft"
    }
    SharpGP2Y0A02YK0F {
      translation -0.055 -0.03 0.09
      rotation -0.5773509358560258 0.577349935856137 -0.577349935856137 -2.094395307179586
      name "irright"
    }
    DEF Arm_R HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.045 -0.01 0.115
      }
      device [
        RotationalMotor {
          name "Arm_R"
        }
      ]
      endPoint Solid {
        translation -0.045 -0.0100006 0.115
        children [
          DEF Arm_end Solid {
            translation 0.01 -0.005 0.035
            children [
              DEF Arm_end Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                }
                geometry Box {
                  size 0.01 0.02 0.01
                }
              }
            ]
            boundingObject USE Arm_end
            physics Physics {
            }
          }
          DEF Arm Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
            }
            geometry Box {
              size 0.01 0.01 0.08
            }
          }
        ]
        name "solid(5)"
        boundingObject USE Arm
        physics Physics {
        }
      }
    }
    DEF Arm_L HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.045 -0.01 0.115
      }
      device [
        RotationalMotor {
          name "Arm_L"
        }
      ]
      endPoint Solid {
        translation 0.045 -0.0100006 0.115
        children [
          DEF Arm_end Solid {
            translation -0.01 -0.005 0.035
            children [
              DEF Arm_end Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                }
                geometry Box {
                  size 0.01 0.02 0.01
                }
              }
            ]
            boundingObject USE Arm_end
            physics Physics {
            }
          }
          DEF Arm Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
            }
            geometry Box {
              size 0.01 0.01 0.08
            }
          }
        ]
        name "solid(4)"
        boundingObject USE Arm
        physics Physics {
        }
      }
    }
    DEF WHEEL_L HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.035 0.02 0.02
      }
      device [
        RotationalMotor {
        }
      ]
      endPoint DEF WHEELL Solid {
        translation 0.05250006474180338 -0.034932601811275235 -0.019999988954602744
        rotation 1.405006846879314e-06 -1.0213688632814558e-06 0.9999999999984913 1.5708038780147195
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
            }
            geometry Cylinder {
              height 0.005
              radius 0.02
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    DEF WHEEL_R HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.035 -0.02 -0.02
      }
      device [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation -0.052500017620600234 -0.03493261127764058 -0.01999997987054661
        rotation 1.6544223715071662e-06 -1.634909414884867e-06 0.999999999997295 1.5707949116739404
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    DEF SUPPORT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0 -0.02 0.06
      }
      device [
        RotationalMotor {
        }
      ]
      endPoint Solid {
        translation 1.0718335617296636e-11 -0.03493996072659426 0.060000007932875285
        rotation 1.6115171442723218e-06 -1.592394465829628e-06 0.9999999999974337 1.5707961649090312
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    DEF BODY Group {
      children [
        DEF Sensor_support Transform {
          translation 0 0.035 0.125
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.667 1 0
              }
              geometry Box {
                size 0.02 0.01 0.05
              }
            }
          ]
        }
        DEF triangle_left Transform {
          translation 0 0 0.025
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 1 0
              }
              geometry IndexedFaceSet {
                coord Coordinate {
                  point [
                    0.015 0.04 0.05
                    0.05 0.04 0.05
                    0.015 0.04 0.08
                  ]
                }
                coordIndex [
                  2, 1, 0, -1
                ]
              }
            }
          ]
        }
        DEF triangle_right Transform {
          translation 0 0 0.025
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 1 0
              }
              geometry IndexedFaceSet {
                coord Coordinate {
                  point [
                    -0.015 0.04 0.05
                    -0.05 0.04 0.05
                    -0.015 0.04 0.08
                  ]
                }
                coordIndex [
                  0, 1, 2
                ]
              }
            }
          ]
        }
        DEF planeL Transform {
          translation 0.0325 0 0.09
          rotation -0.889688808424664 -0.32284393048250826 0.32283993048336956 -1.6874153071795863
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 1 0
              }
              geometry Plane {
                size 0.0461 0.08
              }
            }
          ]
          rotationStep 0.2362
        }
        DEF planeR Transform {
          translation -0.0325 -7.20777e-08 0.09
          rotation -0.25420894236359437 0.6838758449458732 0.6838798449449662 2.64372
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 1 0
              }
              geometry Plane {
                size 0.046098 0.08
              }
            }
          ]
          rotationStep 0.2362
        }
        DEF front Transform {
          translation 0 0 0.09
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.666667 1 0
              }
              geometry Box {
                size 0.03 0.08 0.03
              }
            }
          ]
        }
        DEF main Shape {
          appearance PBRAppearance {
            baseColor 0.666667 1 0
          }
          geometry Box {
            size 0.1 0.08 0.15
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.0251883 0.05 0.0804
      rotation 0 1 0 -1.047195307179586
      children [
        Solid {
          children [
            DEF HC-SR504 Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.015 0.02 0.045
              }
            }
          ]
          boundingObject USE HC-SR504
        }
      ]
      name "us_left"
      boundingObject USE HC-SR504
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.029 0 0
        0.03 174 0.1
        3.99 23942 0.00075
        4 38000 0
      ]
      type "sonar"
      numberOfRays 5
      aperture 0.5236
    }
    DistanceSensor {
      translation -0.0251885 0.05 0.0804
      rotation 0 1 0 -2.0944003061004253
      children [
        Solid {
          children [
            USE HC-SR504
          ]
        }
      ]
      name "us_right"
      boundingObject USE HC-SR504
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.029 0 0
        0.03 174 0.1
        3.99 23142 0.00075
        4 38000 0
      ]
      type "sonar"
      numberOfRays 5
      aperture 0.5236
    }
    LightSensor {
      translation 0 0.029 0.145
      rotation 0.25056115554904 0.25058515556393934 0.9351075805173039 -1.6377453071795864
      children [
        Solid {
          children [
            DEF TEPT4400 Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Capsule {
                height 0.0044
                radius 0.003
              }
            }
          ]
          boundingObject USE TEPT4400
        }
      ]
      name "TEPT4400"
      lookupTable [
        0 0 0
        0.037 0.02 0.1
        0.937 1.89 0.1
        0.989 2 0.1
      ]
      colorFilter 1 0 0
    }
  ]
  contactMaterial "robot"
  boundingObject Box {
    size 0.1 0.08 0.15
  }
  physics Physics {
    density -1
    mass 3
  }
  rotationStep 0.1309
}
Floor {
  translation 0 -0.04 0
  appearance Parquetry {
    type "chequered"
    textureTransform TextureTransform {
    }
  }
}
target {
  translation -0.177495 0.00300549 0.355069
}
}